graphql_compiler/query_formatting/common.py

```diff
@@ -216,7 +250,9 @@ def validate_argument_type(name, expected_type, value):
         )
 
 
-def ensure_arguments_are_provided(expected_types, arguments):
+def ensure_arguments_are_provided(
+    expected_types: Mapping[str, GraphQLType], arguments: Mapping[str, Any]
+) -> None:
     """Ensure that all arguments expected by the query were actually provided."""
```

Comment:
This function was doing more than it promised.

Reply:
Code in macro validation was relying on that. It should use `validate_arguments` now

Reply:
Addressed.